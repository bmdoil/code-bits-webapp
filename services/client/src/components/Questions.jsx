import React, { Component } from 'react';
import AceEditor from 'react-ace';
import 'brace/mode/python';
import 'brace/theme/solarized_dark';

class Questions extends Component {
    constructor(props) {
        super(props);
        this.state = {
            exercises: [],
            editor: {
                value: '# Enter your code here.'
            },
        };
        this.onChange = this.onChange.bind(this);
        this.submitExercise = this.submitExercise.bind(this);
    };
    componentDidMount() {
        this.getQuestions();
    };
    onChange(value) {
        this.setState({
            editor: {
                value: value
            }
        });
    };

    submitExercise(event) {
        event.preventDefault();
    }

    render() {
        return (
            <div>
                <h1 className="title is-1">Questions</h1>
                <hr/><br/>
                {!this.props.isAuthenticated && 
                    <div className="notification is-warning">
                        <span>Register or Login to submit a question.</span>
                    </div>    
                }
                {this.state.exercises.length &&
                    <div key={this.state.exercises[0].id}>
                        <h5 className="title is-5">{this.state.exercises[0].body}</h5>
                        <AceEditor
                            mode="python"
                            theme="solarized_dark"
                            name={(this.state.exercises[0].id).toString()}
                            onLoad={this.onLoad}
                            fontSize={18}
                            height={'350px'}
                            showPrintMargin={true}
                            showGutter={true}
                            highlightActiveLine={true}
                            value={this.state.editor.value}
                            style={{
                                marginBottom: '10px'
                            }}
                            editorProps={{
                                $blockScrolling: Infinity
                            }}
                            onChange={this.onChange}
                        />
                        {this.props.isAuthenticated &&
                            <button className="button is-primary" onClick={this.submitExercise}>Run</button>
                        }

                        <br/><br/>
                    </div>
                }
            </div>
        )
    };

    getQuestions() {
        const exercises = [
            {
                id: 0,
                body: 'Define a function called factorial that takes two integers as arguments and returns the factorial of that number.',
            },
            {
                id: 1,
                body: 'Define a function reverse that takes a string as an argument and returns the string in the reversed order.',
            }
        ];
        this.setState({exercises:exercises});
    }
}

export default Questions;