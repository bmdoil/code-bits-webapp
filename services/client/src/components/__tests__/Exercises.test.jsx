import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import AceEditor from 'react-ace';
jest.mock('react-ace')

import Questions from '../Questions';


test('Questions renders properly', () => {
    const wrapper = shallow(<Questions/>);
    const element = wrapper.find('h5');
    expect(element.length).toBe(1)
});


test('Questions renders snapshot', () => {
    const tree = renderer.create(<Questions/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Questions call componentWillMount', () => {
    const onWillMount = jest.fn();
    Questions.prototype.componentWillMount = onWillMount;
    const wrapper = mount(<Questions/>);
    expect(onWillMount).toHaveBeenCalledTimes(1);
});
