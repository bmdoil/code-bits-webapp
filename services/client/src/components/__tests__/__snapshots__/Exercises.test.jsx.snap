// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercises renders snapshot 1`] = `
<div>
  <h1
    className="title is-1"
  >
    Exercises
  </h1>
  <hr />
  <br />
  <div>
    <h5
      className="title is-5"
    >
      Define a function called factorial that takes two integers as arguments and returns the factorial of that number.
    </h5>
    <button
      className="button is-primary"
    >
      Run
    </button>
    <br />
    <br />
  </div>
</div>
`;
